src/dc_receive_imf.rs

```diff
@@ -482,7 +502,12 @@ async fn add_parts(
 
         if chat_id.is_unset() {
             // try to create a normal chat
-            let create_blocked = if from_id == to_id {
+
+            // When fetching existing messages, we want to show all chats where the user sent a message to.
+            let create_blocked = if from_id == to_id
+                || (fetching_existing_messages && incoming_origin >= Origin::OutgoingBcc)
+                || (incoming_origin >= Origin::AddressBook)
```

Comment:
That's a part that's worth to discuss it. Up to now, we did not create any chats without explicit user interaction. Now, we are creating some at first start.

When I wrote this, I thought: Let's pop up a chat also if the contact is in the address book (`incoming_origin >= Origin::AddressBook`).

Currently, I think that chat messages should always pop up if I the `origin >= OutgoingBcc` (i.e. the user sent a message from another device OR has the sender in the address book) to improve multi-device usage.

For classic emails, I am not sure when they should pop up and when not.

Reply:
> Up to now, we did not create any chats without explicit user interaction. Now, we are creating some at first start.

that's a good point we have not thought about much in previous discussions.

indeed showing chats directly without user interaction might raise the question why _some_ chats do not need confirmation while other do.

as this all is currently only about the use case a user installed Delta Chat for the first time and want to continue a  chat started in a classic mua - maybe just do a contact request as usual.

this would also have the advantage, the user will learn about this mechanism at once.

Reply:
But what do we do when we also show existing classic emails from the inbox, assuming the user said "yes"?

Reply:
classic mail won't show up in contact request after configure - but if, we would do just the same, add them to contact request and ask. also classic email all go through contact requests, they do not pop up automatically.

Reply:
> this would also have the advantage, the user will learn about this mechanism at once.

I think this convinced me.

> but if, we would do just the same, add them to contact request and ask.

Problem is, there are going to be a lot of contact requests then, with the risk that the user just ignores them. So maybe put a message "For starters, I gathered some of your existing emails. You can delete the new chats if you do not want them." 

Anyway, we can discuss this then. For now, I will do as you proposed and mark this resolved.

Just one extra idea: If this is an _existing, outgoing chat message_ then let this pop up, so that if the user reinstalls Delta Chat, the chats are still there (just without the non-decryptable messages). This extra will not have any influence on the first-install behavior.

Reply:
Ok another problem: The fetched existing messages are not shown in the Contact Requests pseudo-chat at the top of the chatlist but you only see them by tapping on "Contact Requests" in the menu. Which a new user won't do.
Possibilities:
- Just mark all messages in deaddrop as "InFresh" during fetch_existing_msgs. That's actually a hack because they will often already be marked as "seen" on the server and are not at all fresh.
- Ok I don't see anything else maybe I'll just do that

Reply:
> Problem is, there are going to be a lot of contact requests then

again, it is mainly about users installing delta chat for the first time, there will typically only once contact request.

for further installations, chats, that the user replied to are no contact requests (at least we could make it that way, not totally sure if this is the current state)

and, in general, contact requests has to be improved independently of this issue. if "show all emails" is enabled also for  everyday use, this stacking is not optimal (one reason we should not push users using "show all emails" currently). a simple idea might be to open the "contact request chat" when tapping on the contact request, it was like that some time ago, however,  improving this is out of scope of this pr.

> Just mark all messages in deaddrop as "InFresh" during fetch_existing_msgs

i also do not see a simple alternative currently. but that might work. we already forward the flag, so it should be easy to do. looking forward to test that out :)

Reply:
Ok, finally done and everything works.